pkg/controller/jobframework/reconciler.go

```diff
@@ -569,34 +569,60 @@ func (r *JobReconciler) recordAdmissionCheckUpdate(wl *kueue.Workload, job Gener
 	}
 }
 
-// IsParentJobManaged checks whether the parent job is managed by kueue.
-func (r *JobReconciler) IsParentJobManaged(ctx context.Context, jobObj client.Object, namespace string) (bool, error) {
-	owner := metav1.GetControllerOf(jobObj)
-	if owner == nil {
-		return false, nil
-	}
-	parentJob := GetEmptyOwnerObject(owner)
-	if parentJob == nil {
-		return false, fmt.Errorf("workload owner %v: %w", owner, ErrUnknownWorkloadOwner)
-	}
-	if err := r.client.Get(ctx, client.ObjectKey{Name: owner.Name, Namespace: namespace}, parentJob); err != nil {
-		return false, errors.Join(ErrWorkloadOwnerNotFound, err)
+// IsAncestorJobManaged checks whether an ancestor job is managed by kueue.
+func (r *JobReconciler) IsAncestorJobManaged(ctx context.Context, jobObj client.Object, namespace string) (bool, error) {
+	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj, namespace)
+	if err != nil {
+		return false, err
 	}
-	return QueueNameForObject(parentJob) != "", nil
+	return ancestor != nil, nil
 }
 
-func (r *JobReconciler) getParentWorkload(ctx context.Context, object client.Object) (*kueue.Workload, error) {
-	owner := metav1.GetControllerOf(object)
-	if owner == nil {
+// getAncestorWorkload returns the Workload object of the Kueue-managed ancestor job.
+func (r *JobReconciler) getAncestorWorkload(ctx context.Context, jobObj client.Object, namespace string) (*kueue.Workload, error) {
+	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj, namespace)
+	if err != nil {
+		return nil, err
+	}
+	if ancestor == nil {
 		return nil, nil
 	}
```

Comment:
```suggestion
	if err != nil || ancestor == nil{
		return nil, err
	}
```

nit: We can minimize this logic.

Reply:
thanks.  folded into #4300.

Reply:
Thank you!